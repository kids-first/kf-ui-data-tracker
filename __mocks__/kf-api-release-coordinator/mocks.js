import {GET_STUDY_RELEASES, GET_RELEASED_STUDY} from '../../src/state/queries';
import getStudyReleases from './responses/getStudyReleases.json';
import getStudyReleasesEmpty from './responses/getStudyReleasesEmpty.json';
import allReleaseStudies from './responses/allReleaseStudies.json';

export const coordMocks = {
  getStudyReleases: {
    request: {
      query: GET_STUDY_RELEASES,
      variables: {
        id: 'U3R1ZHlOb2RlOlNEXzhXWDhRUTA2',
      },
    },
    result: getStudyReleases,
  },
  getStudyReleasesError: {
    request: {
      query: GET_STUDY_RELEASES,
      variables: {
        id: 'U3R1ZHlOb2RlOlNEXzhXWDhRUTA2',
      },
    },
    error: new Error('Failed to fetch releases information'),
  },
  getStudyReleasesEmpty: {
    request: {
      query: GET_STUDY_RELEASES,
      variables: {
        id: 'U3R1ZHlOb2RlOlNEXzhXWDhRUTA2',
      },
    },
    result: getStudyReleasesEmpty,
  },
  allReleaseStudies: {
    request: {
      query: GET_RELEASED_STUDY,
    },
    result: allReleaseStudies,
  },
  allReleaseStudiesError: {
    request: {
      query: GET_RELEASED_STUDY,
    },
    error: new Error('Failed to fetch study releases information'),
  },
};
