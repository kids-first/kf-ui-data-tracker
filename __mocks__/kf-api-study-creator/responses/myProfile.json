{
  "data": {
    "myProfile": {
      "id": "VXNlck5vZGU6MTY=",
      "slackNotify": false,
      "slackMemberId": "",
      "studySubscriptions": {"edges": [], "__typename": "StudyNodeConnection"},
      "dateJoined": "2020-03-29T01:36:54.053128+00:00",
      "username": "devadmin",
      "firstName": "",
      "lastName": "",
      "displayName": "devadmin",
      "email": "admin@myproject.com",
      "picture": "",
      "roles": ["ADMIN"],
      "studies": {"edges": [], "__typename": "StudyNodeConnection"},
      "groups": {
        "edges": [
          {
            "node": {
              "id": "R3JvdXBOb2RlOjE=",
              "name": "Administrators",
              "permissions": {
                "edges": [
                  {
                    "node": {
                      "id": "UGVybWlzc2lvbk5vZGU6MTI=",
                      "name": "Can view group",
                      "codename": "view_group",
                      "__typename": "PermissionNode"
                    },
                    "__typename": "PermissionNodeEdge"
                  },
                  {
                    "node": {
                      "id": "UGVybWlzc2lvbk5vZGU6OA==",
                      "name": "Can view permission",
                      "codename": "view_permission",
                      "__typename": "PermissionNode"
                    },
                    "__typename": "PermissionNodeEdge"
                  },
                  {
                    "node": {
                      "id": "UGVybWlzc2lvbk5vZGU6NjE=",
                      "name": "Can link a bucket to a study",
                      "codename": "link_bucket",
                      "__typename": "PermissionNode"
                    },
                    "__typename": "PermissionNodeEdge"
                  },
                  {
                    "node": {
                      "id": "UGVybWlzc2lvbk5vZGU6NjA=",
                      "name": "Can list all buckets",
                      "codename": "list_all_bucket",
                      "__typename": "PermissionNode"
                    },
                    "__typename": "PermissionNodeEdge"
                  },
                  {
                    "node": {
                      "id": "UGVybWlzc2lvbk5vZGU6NjI=",
                      "name": "Can unlink a bucket to a study",
                      "codename": "unlink_bucket",
                      "__typename": "PermissionNode"
                    },
                    "__typename": "PermissionNodeEdge"
                  },
                  {
                    "node": {
                      "id": "UGVybWlzc2lvbk5vZGU6NTk=",
                      "name": "Can view bucket",
                      "codename": "view_bucket",
                      "__typename": "PermissionNode"
                    },
                    "__typename": "PermissionNodeEdge"
                  },
                  {
                    "node": {
                      "id": "UGVybWlzc2lvbk5vZGU6NzY=",
                      "name": "Can view job",
                      "codename": "view_job",
                      "__typename": "PermissionNode"
                    },
                    "__typename": "PermissionNodeEdge"
                  },
                  {
                    "node": {
                      "id": "UGVybWlzc2lvbk5vZGU6Nzg=",
                      "name": "Can view queues",
                      "codename": "view_queue",
                      "__typename": "PermissionNode"
                    },
                    "__typename": "PermissionNodeEdge"
                  },
                  {
                    "node": {
                      "id": "UGVybWlzc2lvbk5vZGU6Nzc=",
                      "name": "Can view settings",
                      "codename": "view_settings",
                      "__typename": "PermissionNode"
                    },
                    "__typename": "PermissionNodeEdge"
                  },
                  {
                    "node": {
                      "id": "UGVybWlzc2lvbk5vZGU6Njk=",
                      "name": "Can change user",
                      "codename": "change_user",
                      "__typename": "PermissionNode"
                    },
                    "__typename": "PermissionNodeEdge"
                  },
                  {
                    "node": {
                      "id": "UGVybWlzc2lvbk5vZGU6NzI=",
                      "name": "Can list all users",
                      "codename": "list_all_user",
                      "__typename": "PermissionNode"
                    },
                    "__typename": "PermissionNodeEdge"
                  },
                  {
                    "node": {
                      "id": "UGVybWlzc2lvbk5vZGU6NjY=",
                      "name": "Can view event",
                      "codename": "view_event",
                      "__typename": "PermissionNode"
                    },
                    "__typename": "PermissionNodeEdge"
                  },
                  {
                    "node": {
                      "id": "UGVybWlzc2lvbk5vZGU6Mzc=",
                      "name": "Can list all versions",
                      "codename": "list_all_version",
                      "__typename": "PermissionNode"
                    },
                    "__typename": "PermissionNodeEdge"
                  },
                  {
                    "node": {
                      "id": "UGVybWlzc2lvbk5vZGU6Mjk=",
                      "name": "Can add download token",
                      "codename": "add_downloadtoken",
                      "__typename": "PermissionNode"
                    },
                    "__typename": "PermissionNodeEdge"
                  },
                  {
                    "node": {
                      "id": "UGVybWlzc2lvbk5vZGU6MzE=",
                      "name": "Can delete download token",
                      "codename": "delete_downloadtoken",
                      "__typename": "PermissionNode"
                    },
                    "__typename": "PermissionNodeEdge"
                  },
                  {
                    "node": {
                      "id": "UGVybWlzc2lvbk5vZGU6MzI=",
                      "name": "Can view download token",
                      "codename": "view_downloadtoken",
                      "__typename": "PermissionNode"
                    },
                    "__typename": "PermissionNodeEdge"
                  },
                  {
                    "node": {
                      "id": "UGVybWlzc2lvbk5vZGU6MjE=",
                      "name": "Can add file",
                      "codename": "add_file",
                      "__typename": "PermissionNode"
                    },
                    "__typename": "PermissionNodeEdge"
                  },
                  {
                    "node": {
                      "id": "UGVybWlzc2lvbk5vZGU6MjI=",
                      "name": "Can change file",
                      "codename": "change_file",
                      "__typename": "PermissionNode"
                    },
                    "__typename": "PermissionNodeEdge"
                  },
                  {
                    "node": {
                      "id": "UGVybWlzc2lvbk5vZGU6MjM=",
                      "name": "Can delete file",
                      "codename": "delete_file",
                      "__typename": "PermissionNode"
                    },
                    "__typename": "PermissionNodeEdge"
                  },
                  {
                    "node": {
                      "id": "UGVybWlzc2lvbk5vZGU6MjU=",
                      "name": "Can list all files",
                      "codename": "list_all_file",
                      "__typename": "PermissionNode"
                    },
                    "__typename": "PermissionNodeEdge"
                  },
                  {
                    "node": {
                      "id": "UGVybWlzc2lvbk5vZGU6MjQ=",
                      "name": "Can view file",
                      "codename": "view_file",
                      "__typename": "PermissionNode"
                    },
                    "__typename": "PermissionNodeEdge"
                  },
                  {
                    "node": {
                      "id": "UGVybWlzc2lvbk5vZGU6Mzg=",
                      "name": "Can add version",
                      "codename": "add_version",
                      "__typename": "PermissionNode"
                    },
                    "__typename": "PermissionNodeEdge"
                  },
                  {
                    "node": {
                      "id": "UGVybWlzc2lvbk5vZGU6Mzk=",
                      "name": "Can change version",
                      "codename": "change_version",
                      "__typename": "PermissionNode"
                    },
                    "__typename": "PermissionNodeEdge"
                  },
                  {
                    "node": {
                      "id": "UGVybWlzc2lvbk5vZGU6NDA=",
                      "name": "Can delete version",
                      "codename": "delete_version",
                      "__typename": "PermissionNode"
                    },
                    "__typename": "PermissionNodeEdge"
                  },
                  {
                    "node": {
                      "id": "UGVybWlzc2lvbk5vZGU6NDE=",
                      "name": "Can view version",
                      "codename": "view_version",
                      "__typename": "PermissionNode"
                    },
                    "__typename": "PermissionNodeEdge"
                  },
                  {
                    "node": {
                      "id": "UGVybWlzc2lvbk5vZGU6NTA=",
                      "name": "Can add a collaborator to the study",
                      "codename": "add_collaborator",
                      "__typename": "PermissionNode"
                    },
                    "__typename": "PermissionNodeEdge"
                  },
                  {
                    "node": {
                      "id": "UGVybWlzc2lvbk5vZGU6NTE=",
                      "name": "Can remove a collaborator to the study",
                      "codename": "remove_collaborator",
                      "__typename": "PermissionNode"
                    },
                    "__typename": "PermissionNodeEdge"
                  },
                  {
                    "node": {
                      "id": "UGVybWlzc2lvbk5vZGU6NDg=",
                      "name": "Can view study",
                      "codename": "view_study",
                      "__typename": "PermissionNode"
                    },
                    "__typename": "PermissionNodeEdge"
                  }
                ],
                "__typename": "PermissionNodeConnection"
              },
              "__typename": "GroupNode"
            },
            "__typename": "GroupNodeEdge"
          }
        ],
        "__typename": "GroupNodeConnection"
      },
      "__typename": "UserNode"
    }
  }
}
