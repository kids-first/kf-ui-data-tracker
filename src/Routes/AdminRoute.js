import React from 'react';
import jwtDecode from 'jwt-decode';
import {Route, Redirect, withRouter} from 'react-router-dom';
import {Amplitude} from '@amplitude/react-amplitude';

export const isAdmin = () => {
  const token = localStorage.getItem('accessToken');
  if (token == null) {
    return false;
  }

  const decoded = jwtDecode(token);
  if (decoded.exp < Math.floor(new Date().getTime() / 1000)) return false;

  // Extract roles based on whether the token is from auth0
  const roles = decoded.context
    ? decoded.context.user.roles
    : decoded['https://kidsfirstdrc.org/roles'];

  return roles.includes('ADMIN');
};

/**
 * A route that only allows access to admin users, otherwise routes to the
 * index page
 */
const AdminRoute = ({component: Component, path, ...rest}) => {
  let comp = <Component />;

  return (
    <Amplitude
      eventProperties={{
        view: comp.type.WrappedComponent
          ? comp.type.WrappedComponent.name
          : comp.type.name,
        route: path,
      }}
    >
      <Route
        {...rest}
        path={path}
        render={props =>
          isAdmin() ? <Component {...props} /> : <Redirect to="/" />
        }
      />
    </Amplitude>
  );
};

export default withRouter(AdminRoute);
