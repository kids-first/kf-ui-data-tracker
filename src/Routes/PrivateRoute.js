import React from 'react';
import jwtDecode from 'jwt-decode';
import {Route, Redirect, withRouter} from 'react-router-dom';
import {Amplitude} from '@amplitude/react-amplitude';

export const hasToken = () => {
  const token = localStorage.getItem('accessToken');
  if (token == null) {
    return false;
  }
  return jwtDecode(token).exp > Math.floor(new Date().getTime() / 1000);
};

const PrivateRoute = ({component: Component, ...rest}) => {
  // note: unsure if this mounts the
  let comp = <Component />;

  return (
    <Amplitude
      eventProperties={{
        view: comp.type.WrappedComponent
          ? comp.type.WrappedComponent.name
          : comp.type.name,
        route: rest.path,
      }}
    >
      <Route
        {...rest}
        render={props =>
          hasToken() ? (
            <Component {...props} />
          ) : (
            <Redirect
              to={{
                pathname: '/login',
                state: {from: window.location.pathname},
              }}
            />
          )
        }
      />
    </Amplitude>
  );
};

export default withRouter(PrivateRoute);
