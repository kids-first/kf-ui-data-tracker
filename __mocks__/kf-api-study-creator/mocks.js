import {
  ALL_STUDIES,
  GET_STUDY_BY_ID,
  MY_PROFILE,
  ALL_EVENTS,
  ALL_USERS,
  GET_PROJECTS,
  STATUS,
} from '../../src/state/queries';
import {
  CREATE_PROJECT,
  SYNC_PROJECTS,
  CREATE_STUDY,
  UPDATE_STUDY,
} from '../../src/state/mutations';
import {GET_FILE_BY_ID} from '../../src/documents/queries';
import {
  DELETE_FILE,
  UPDATE_FILE,
  UPDATE_VERSION,
  CREATE_VERSION,
  FILE_DOWNLOAD_URL,
} from '../../src/documents/mutations';
import {GET_DEV_TOKENS} from '../../src/admin/queries';
import {DELETE_DEV_TOKEN, CREATE_DEV_TOKEN} from '../../src/admin/mutations';
import allStudies from './responses/allStudies';
import allStudies_empty from './responses/allStudies_empty';
import studyByKfId from './responses/studyByKfId.json';
import studyByKfId_refetch from './responses/studyByKfId_refetch.json';
import studyByKfId_event_err from './responses/studyByKfId_event_err.json';
import studyByKfId_no_projects from './responses/studyByKfId_no_projects.json';
import studyByKfId_empty from './responses/studyByKfId_empty.json';
import deleteFile from './responses/deleteFile.json';
import fileByKfId from './responses/fileByKfId.json';
import fileByKfId_empty from './responses/fileByKfId_empty.json';
import myProfile from './responses/myProfile.json';
import updateFile from './responses/updateFile.json';
import updateVersion from './responses/updateVersion.json';
import createVersion from './responses/createVersion.json';
import createStudy from './responses/createStudy.json';
import updateStudy from './responses/updateStudy.json';
import devTokens from './responses/devTokens.json';
import createDevToken from './responses/createDevToken.json';
import allEvents from './responses/allEvents.json';
import allEvents_refetch from './responses/allEvents_refetch.json';
import allEvents_20 from './responses/allEvents_20.json';
import allEvents_40 from './responses/allEvents_40.json';
import allUsers from './responses/allUsers.json';
import allProjects from './responses/allProjects.json';
import signedUrl from './responses/signedUrl.json';
import syncProjects from './responses/syncProjects.json';
import deleteToken from './responses/deleteToken.json';
import status from './responses/status.json';
import allEvents_studyCreation from './responses/allEvents_studyCreation.json';
import allEvents_studyCreation_err from './responses/allEvents_studyCreation_err.json';
import allEvents_studyCreation_ing from './responses/allEvents_studyCreation_ing.json';

export const mocks = [
  {
    request: {
      query: ALL_STUDIES,
    },
    result: allStudies,
  },
  {
    request: {
      query: GET_STUDY_BY_ID,
      variables: {
        kfId: 'SD_8WX8QQ06',
      },
    },
    result: studyByKfId,
  },
  {
    request: {
      query: GET_STUDY_BY_ID,
      variables: {
        kfId: 'SD_8WX8QQ06',
      },
    },
    error: new Error('something went wrong'),
  },
  {
    request: {
      query: DELETE_FILE,
      variables: {
        kfId: 'SF_Y07IN1HO',
      },
    },
    result: deleteFile,
  },
  {
    request: {
      query: GET_FILE_BY_ID,
      variables: {
        kfId: 'SF_5ZPEM167',
      },
    },
    result: fileByKfId,
  },
  {
    request: {
      query: UPDATE_FILE,
      variables: {
        kfId: 'SF_5ZPEM167',
        name: 'foo bar file',
        description: 'Some description here',
        fileType: 'CLN',
      },
    },
    result: updateFile,
  },
  {
    request: {
      query: UPDATE_VERSION,
      variables: {
        versionId: 'FV_DMTKEFQW',
        description: 'version description',
        state: 'APP',
      },
    },
    result: updateVersion,
  },
  {
    request: {
      query: CREATE_VERSION,
      variables: {
        file: {
          lastModified: 1567797130510,
          name: 'organize.tsv',
          size: 7,
          type: 'text/tab-separated-values',
          webkitRelativePath: '',
        },
        fileId: 'SF_5ZPEM167',
        description: 'version description',
      },
    },
    result: createVersion,
  },
  {
    request: {
      query: MY_PROFILE,
    },
    result: myProfile,
  },
  {
    request: {
      query: GET_DEV_TOKENS,
    },
    result: devTokens,
  },
  {
    request: {
      query: CREATE_DEV_TOKEN,
      variables: {
        name: 'my token 4',
      },
    },
    result: createDevToken,
  },
  {
    request: {
      query: ALL_EVENTS,
      variables: {
        studyId: 'SD_8WX8QQ06',
        orderBy: '-created_at',
        first: 20,
      },
    },
    result: allEvents,
  },
  {
    request: {
      query: ALL_EVENTS,
      variables: {
        studyId: 'SD_8WX8QQ06',
        orderBy: '-created_at',
        first: 20,
      },
    },
    error: new Error('something went wrong with your event logs request'),
  },
  {
    request: {
      query: ALL_USERS,
    },
    result: allUsers,
  },
  {
    request: {
      query: MY_PROFILE,
    },
    error: new Error('Failed to get user profile information'),
  },
  {
    request: {
      query: GET_PROJECTS,
      variables: {
        study: '',
        deleted: false,
      },
    },
    result: allProjects,
  },
  {
    request: {
      query: GET_PROJECTS,
      variables: {
        study: '',
        deleted: false,
      },
    },
    error: new Error('Failed to get the Cavatica project data for this study'),
  },
  {
    request: {
      query: GET_PROJECTS,
      variables: {
        study: null,
        deleted: null,
      },
    },
    result: allProjects,
  },
  {
    request: {
      query: GET_PROJECTS,
      variables: {},
    },
    result: allProjects,
  },
  {
    request: {
      query: SYNC_PROJECTS,
    },
    result: syncProjects,
  },
  {
    request: {
      query: SYNC_PROJECTS,
    },
    error: new Error('Failed to sync Cavatica projects'),
  },
  {
    request: {
      query: CREATE_STUDY,
      variables: {
        workflows: ['bwa_mem', 'gatk_haplotypecaller'],
        input: {
          externalId: 'benchmark extensible e-business',
          name: 'benchmark extensible e-business',
          shortName: '',
          description: '',
          releaseDate: null,
          version: '',
          attribution: '',
          anticipatedSamples: 0,
          awardeeOrganization: '',
        },
      },
    },
    result: createStudy,
  },
  {
    request: {
      query: CREATE_STUDY,
      variables: {
        workflows: ['bwa_mem', 'gatk_haplotypecaller'],
        input: {
          externalId: 'benchmark extensible e-business',
          name: 'benchmark extensible e-business',
          shortName: '',
          description: '',
          releaseDate: null,
          version: '',
          attribution: '',
          anticipatedSamples: 0,
          awardeeOrganization: '',
        },
      },
    },
    error: new Error('Failed to create the new study'),
  },
  {
    request: {
      query: UPDATE_STUDY,
      variables: {
        id: 'U3R1ZHlOb2RlOlNEXzhXWDhRUTA2',
        input: {
          externalId: 'test_study_external_id',
          name: 'benchmark extensible e-business',
          shortName: '',
          description:
            'Sudy description in markdown, commonly the X01 abstract text. --  As test study description',
          releaseDate: '2019-08-24',
          anticipatedSamples: 100,
          awardeeOrganization: 'CHOP',
          attribution: '',
          version: '',
          bucket: 'stay-wrong-ready',
        },
      },
    },
    result: updateStudy,
  },
  {
    request: {
      query: UPDATE_STUDY,
      variables: {
        id: 'U3R1ZHlOb2RlOlNEXzhXWDhRUTA2',
        input: {
          externalId: 'test_study_external_id',
          name: 'benchmark extensible e-business',
          shortName: '',
          description:
            'Sudy description in markdown, commonly the X01 abstract text. --  As test study description',
          releaseDate: '2019-08-24',
          anticipatedSamples: 100,
          awardeeOrganization: 'CHOP',
          attribution: '',
          version: '',
          bucket: 'stay-wrong-ready',
        },
      },
    },
    error: new Error('Failed to update the study'),
  },
  {
    request: {
      query: ALL_EVENTS,
      variables: {
        orderBy: '-created_at',
        first: 20,
      },
    },
    result: allEvents,
  },
  {
    request: {
      query: GET_STUDY_BY_ID,
      variables: {
        kfId: 'SD_8WX8QQ06',
      },
    },
    result: studyByKfId_refetch,
  },
  {
    request: {
      query: CREATE_VERSION,
      variables: {
        file: {},
        fileId: 'SF_5ZPEM167',
        description: 'my changes',
      },
    },
    result: createVersion,
  },
  {
    request: {
      query: ALL_EVENTS,
      variables: {
        orderBy: '-created_at',
        username: 'Justin Heath',
        first: 20,
      },
    },
    result: allEvents_refetch,
  },
  {
    request: {
      query: ALL_EVENTS,
      variables: {
        orderBy: '-created_at',
        username: 'Justin Heath',
        studyId: 'SD_SG5N41K8',
        first: 20,
      },
    },
    result: allEvents_refetch,
  },
  {
    request: {
      query: ALL_EVENTS,
      variables: {
        orderBy: '-created_at',
        username: 'Justin Heath',
        studyId: 'SD_SG5N41K8',
        eventType: 'FV_CRE',
        first: 20,
      },
    },
    result: allEvents_refetch,
  },
  {
    request: {
      query: ALL_EVENTS,
      variables: {
        orderBy: '-created_at',
        first: 20,
      },
    },
    error: new Error('Failed to fetch events information'),
  },
  {
    request: {
      query: FILE_DOWNLOAD_URL,
      variables: {
        studyId: 'SD_00000000',
        fileId: 'SF_5ZPEM167',
        versionId: 'FV_TT8NF09R',
      },
    },
    result: signedUrl,
  },
  {
    request: {
      query: GET_PROJECTS,
      variables: {},
    },
    error: new Error('Failed to fetch projects information'),
  },
  {
    request: {
      query: DELETE_DEV_TOKEN,
      variables: {
        name: 'my token 1',
      },
    },
    result: deleteToken,
  },
  {
    request: {
      query: CREATE_DEV_TOKEN,
      variables: {
        name: 'my token 4',
      },
    },
    error: new Error('Failed to create new token'),
  },
  {
    request: {
      query: GET_DEV_TOKENS,
    },
    error: new Error('Failed to fetch tokens information'),
  },
  {
    request: {
      query: ALL_EVENTS,
      variables: {
        orderBy: '-created_at',
        studyId: 'SD_8WX8QQ06',
        eventType: 'FV_CRE',
        first: 20,
      },
    },
    result: allEvents_refetch,
  },
  {
    request: {
      query: ALL_EVENTS,
      variables: {
        orderBy: '-created_at',
        studyId: 'SD_8WX8QQ06',
      },
    },
    result: allEvents_studyCreation,
  },
  {
    request: {
      query: ALL_EVENTS,
      variables: {
        orderBy: '-created_at',
        studyId: 'SD_8WX8QQ06',
      },
    },
    result: allEvents_studyCreation_err,
  },
  {
    request: {
      query: ALL_EVENTS,
      variables: {
        orderBy: '-created_at',
        studyId: 'SD_8WX8QQ06',
      },
    },
    result: allEvents_studyCreation_ing,
  },
  {
    request: {
      query: ALL_EVENTS,
      variables: {
        studyId: 'SD_8WX8QQ06',
        orderBy: '-created_at',
      },
    },
    error: new Error('Failed to fetch events information'),
  },
  {
    request: {
      query: STATUS,
    },
    result: status,
  },
  {
    request: {
      query: GET_STUDY_BY_ID,
      variables: {
        kfId: 'SD_8WX8QQ06',
      },
    },
    result: studyByKfId_event_err,
  },
  {
    request: {
      query: GET_STUDY_BY_ID,
      variables: {
        kfId: 'SD_8WX8QQ06',
      },
    },
    result: studyByKfId_no_projects,
  },
  {
    request: {
      query: ALL_EVENTS,
      variables: {orderBy: '-created_at', first: 20},
    },
    result: allEvents_20,
  },
  {
    request: {
      query: ALL_EVENTS,
      variables: {
        orderBy: '-created_at',
        first: 20,
        cursor: 'YXJyYXljb25uZWN0aW9uOjE5',
      },
    },
    result: allEvents_40,
  },
  {
    request: {
      query: ALL_EVENTS,
      variables: {
        studyId: 'SD_8WX8QQ06',
        orderBy: '-created_at',
        first: 20,
      },
    },
    result: allEvents_20,
  },
  {
    request: {
      query: ALL_EVENTS,
      variables: {
        studyId: 'SD_8WX8QQ06',
        orderBy: '-created_at',
        first: 20,
        cursor: 'YXJyYXljb25uZWN0aW9uOjE5',
      },
    },
    result: allEvents_40,
  },
  {
    request: {
      query: ALL_STUDIES,
    },
    error: new Error('Failed to fetch study information'),
  },
  {
    request: {
      query: ALL_STUDIES,
    },
    result: allStudies_empty,
  },
  {
    request: {
      query: GET_STUDY_BY_ID,
      variables: {
        kfId: 'SD_INVALID',
      },
    },
    result: studyByKfId_empty,
  },
  {
    request: {
      query: GET_FILE_BY_ID,
      variables: {
        kfId: 'SF_INVALIDS',
      },
    },
    result: fileByKfId_empty,
  },
  {
    request: {
      query: ALL_EVENTS,
      variables: {
        studyId: 'SD_INVALID',
        orderBy: '-created_at',
        first: 20,
      },
    },
    result: allEvents,
  },
  {
    request: {
      query: CREATE_STUDY,
      variables: {
        input: {
          name: 'benchmark extensible e-business',
          shortName: '',
          description: '',
          externalId: 'benchmark-extensible-e-business',
          collaborators: ['VXNlck5vZGU6MQ==', 'VXNlck5vZGU6Me=='],
        },
        workflows: [],
      },
    },
    result: createStudy,
  },
  {
    request: {
      query: CREATE_STUDY,
      variables: {
        workflows: [],
        input: {
          externalId: 'benchmark-extensible-e-business',
          name: 'benchmark extensible e-business',
          shortName: '',
          description: '',
          collaborators: [],
        },
      },
    },
    error: new Error('Failed to create the new research study'),
  },
  {
    request: {
      query: UPDATE_STUDY,
      variables: {
        id: 'U3R1ZHlOb2RlOlNEXzhXWDhRUTA2',
        input: {
          name: 'benchmark extensible e-business',
          shortName: 'Test short name input on update',
          description:
            'Sudy description in markdown, commonly the X01 abstract text. --  As test study description',
          collaborators: ['VXNlck5vZGU6MQ==', 'VXNlck5vZGU6Me=='],
        },
      },
    },
    result: updateStudy,
  },
  {
    request: {
      query: UPDATE_STUDY,
      variables: {
        id: 'U3R1ZHlOb2RlOlNEXzhXWDhRUTA2',
        input: {
          name: 'benchmark extensible e-business',
          shortName: '',
          description:
            'Sudy description in markdown, commonly the X01 abstract text. --  As test study description',
          collaborators: ['VXNlck5vZGU6MQ==', 'VXNlck5vZGU6Me=='],
        },
      },
    },
    error: new Error('Failed to update the study'),
  },
  {
    request: {
      query: ALL_USERS,
    },
    error: new Error('Failed to fetch users information'),
  },
];
