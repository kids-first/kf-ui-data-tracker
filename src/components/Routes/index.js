import React from 'react';
import {BrowserRouter as Router, Route, Switch} from 'react-router-dom';
import PrivateRoute from './PrivateRoute';

import {
  LoginView,
  InvestigatorListView,
  FileUploadView,
  FileListView,
} from '../../views';

const Routes = () => (
  <Router>
    <Switch>
      <Route path="/login" component={LoginView} />
      <PrivateRoute exact path="/" component={InvestigatorListView} />
      <PrivateRoute path="/file-upload" component={FileUploadView} />
      <PrivateRoute path="/file-list" component={FileListView} />
    </Switch>
  </Router>
);

export default Routes;
