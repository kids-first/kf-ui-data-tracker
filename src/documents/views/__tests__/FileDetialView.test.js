import React from 'react';
import wait from 'waait';
import {MockedProvider} from '@apollo/react-testing';
import {MemoryRouter} from 'react-router-dom';
import {render} from '@testing-library/react';
import Routes from '../../../Routes';
import {mocks} from '../../../../__mocks__/kf-api-study-creator/mocks';
import myProfile from '../../../../__mocks__/kf-api-study-creator/responses/myProfile.json';

jest.mock('auth0-js');

it('render file detail view with an error', async () => {
  const tree = render(
    <MockedProvider
      resolvers={{
        Query: {
          myProfile: _ => myProfile.data.myProfile,
        },
      }}
      mocks={[mocks[1], mocks[8], mocks[69]]}
    >
      <MemoryRouter
        initialEntries={['/study/SD_8WX8QQ06/documents/SF_5ZPEMOOO']}
      >
        <Routes />
      </MemoryRouter>
    </MockedProvider>,
  );
  await wait();
  expect(tree.container).toMatchSnapshot();
  expect(
    tree.queryByText(/Failed to get the file information/i),
  ).not.toBeNull();
});
