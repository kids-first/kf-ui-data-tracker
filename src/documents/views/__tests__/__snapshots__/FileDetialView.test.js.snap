// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render file detail view with an error 1`] = `
<div>
  <div
    class="ui large top attached menu"
  >
    <div
      class="ui container"
    >
      <div
        class="item"
      >
        <img
          alt="Kids First logo"
          src="test-file-stub"
        />
      </div>
      <a
        aria-current="page"
        class="header item"
        href="/"
      >
        Data Tracker
      </a>
      <a
        class="item"
        href="/"
      >
        Studies
      </a>
      <div
        class="right menu"
      >
        <div
          aria-expanded="false"
          class="ui dropdown link item"
          role="listbox"
          tabindex="0"
        >
          Admin
          <i
            aria-hidden="true"
            class="dropdown icon"
          />
          <div
            class="menu transition"
          >
            <a
              class="item"
              href="/configuration"
              role="option"
            >
              <i
                aria-hidden="true"
                class="settings icon"
              />
              Configuration
            </a>
            <a
              class="item"
              href="/buckets"
              role="option"
            >
              <i
                aria-hidden="true"
                class="trash icon"
              />
              Buckets
            </a>
            <a
              class="item"
              href="/tokens"
              role="option"
            >
              <i
                aria-hidden="true"
                class="key icon"
              />
              Developer Tokens
            </a>
            <a
              class="item"
              href="/cavatica-projects"
              role="option"
            >
              <i
                aria-hidden="true"
                class="folder open icon"
              />
              Cavatica Projects
            </a>
            <a
              class="item"
              href="/events"
              role="option"
            >
              <i
                aria-hidden="true"
                class="history icon"
              />
              Event Log
            </a>
            <a
              class="item"
              href="/users"
              role="option"
            >
              <i
                aria-hidden="true"
                class="users icon"
              />
              Users
            </a>
          </div>
        </div>
        <div
          aria-expanded="false"
          class="ui dropdown link item"
          role="listbox"
          tabindex="0"
        >
          <img
            alt="bendolly"
            class="ui avatar image"
            src="https://lh3.googleusercontent.com/-7Agefge42sw/AAAAAAAAAAI/AAAAAAAAAAA/ACHi3reWt8KI4nO_arpKCb4FzokaKnLgtg/mo/photo.jpg"
          />
          bendolly
          <i
            aria-hidden="true"
            class="dropdown icon"
          />
          <div
            class="menu transition"
          >
            <a
              class="item"
              href="/profile"
              role="option"
            >
              <i
                aria-hidden="true"
                class="user icon"
              />
              Profile
            </a>
            <a
              class="item"
              href="/logout"
              role="option"
            >
              <i
                aria-hidden="true"
                class="log out icon"
              />
              Logout
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="page"
  >
    <section
      id="study"
    >
      <div
        class="ui basic secondary segment"
      >
        <div
          class="ui container"
        >
          <h1
            class="ui header"
          >
            benchmark extensible e-business
          </h1>
          <button
            class="ui tiny basic icon button"
          >
            <i
              aria-hidden="true"
              class="copy icon"
            />
            SD_8WX8QQ06
          </button>
        </div>
      </div>
      <div
        class="ui container"
      >
        <div
          class="ui pink pointing secondary menu"
        >
          <a
            class="item"
            href="/study/SD_8WX8QQ06/basic-info/info"
          >
            Basic Info
          </a>
          <a
            aria-current="page"
            class="active item active"
            href="/study/SD_8WX8QQ06/documents"
          >
            Documents
          </a>
          <a
            class="item"
            href="/study/SD_8WX8QQ06/cavatica"
          >
            Cavatica
          </a>
          <a
            class="item"
            href="/study/SD_8WX8QQ06/collaborators"
          >
            Collaborators
          </a>
          <a
            class="item"
            href="/study/SD_8WX8QQ06/logs"
          >
            Logs
          </a>
          <a
            class="item"
            href="/study/SD_8WX8QQ06/releases"
          >
            Releases
            <div
              class="ui blue mini label"
            >
              BETA
            </div>
          </a>
        </div>
      </div>
    </section>
    <div
      class="ui basic vertical segment ui container"
    >
      <div
        class="ui icon negative message"
      >
        <i
          aria-hidden="true"
          class="warning circle icon"
        />
        <div
          class="content"
        >
          <div
            class="header"
          >
            Error
          </div>
          <p>
            Network error: Failed to get the file information
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    class="ui segment footer"
  >
    <div
      class="ui container text-10"
    >
      <div
        class="ui horizontal list"
        role="list"
      >
        <div
          class="item"
          role="listitem"
        >
          <div
            class="ui basic horizontal label text-10"
          />
        </div>
        <div
          class="item"
          role="listitem"
        >
          UI
           
          <a
            href="https://github.com/kids-first/kf-ui-data-tracker/commit/undefined"
            rel="noopener noreferrer"
            target="_blank"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
